src/rowtable.typ

```diff
@@ -278,15 +279,20 @@
 /// - table (function): Table function to use to build the final table. Intended for use with
 ///   table wrappers from other packages. (The function ```typst arguments``` can be used for
 ///   argument pass-through.)
+/// - cell-function (function): Cell function to use (either `table.cell` or `grid.cell`). You normally do not need to specify this argument.
 /// -> table, content
-#let rowtable(..args, separator: "&", separator-eq: auto, row-filler: none, column-width: none, table: std.table) = {
+#let rowtable(..args, separator: "&", separator-eq: auto, row-filler: none, column-width: none, table: std.table, cell-function: auto) = {
   // type check parameters
   assert(type(separator) == str, message: "Separator must be string")
   assert(
     separator-eq == none or
     separator-eq == auto or
     type(separator-eq) == content and isfuncv(separator-eq, math.equation, align-point, mathsymbol),
     message: "separator-eq must be none, auto, & or math symbol")
+  let cell-function = if cell-function == auto {
+    if table == std.grid { std.grid.cell } else { std.table.cell }
```

Comment:
You may want to make the table / grid choice a separate parameter, as you may want to use a wrapper that produces a grid

Reply:
well, i suppose my proposal is isomorphic to your cell-function parameter so that's probably fine

Reply:
Maybe I don't understand exactly what it is. If the user picks between rowtable and rowgrid correctly, they never need to use the `cell-function` parameter, it is kind of an internal crutch.

Is the suggestion that instead of `cell-function` the parameter is `kind: table | grid` ?